### ISSUE TITLE
LG C8 Airplay 2

### ISSUE THREAD

**Issue by ezerez** [ejurgensen/forked-daapd#1174]
I just installed the newest build of forked-daapd. And the playback on my LG with airplay 2 works great. I just notice that when the TV is off and i select the LG as a source that i get a http 500 error return. If i select it on any of my apple devices as a source when it is off, it turns on the TV. Dont know if this is a bug or something that can be implimented but it would be great to have the posibility that it also turns on the TV. 

thank you for all the hard work!

**Comment by ejurgensen** [comment]
Please set the log level to debug (in /etc/forked-daapd.conf) and try to get some logging of when you select the LG. Don’t cut and paste lengthy log outtakes here on github. Instead, attach the log file.

**Comment by ezerez** [comment]
Thank you for the very quick reply. I added 2 logs. one when the TV is off and I try to select it. and one where I turn on the TV and then select is as a source which works. I also noticed now when i restart forked daapd and the TV is off the listing for the C8 is not there any more. On my apple devices I can still see it and select it.

[debug with lg tv turned off.log](https://github.com/ejurgensen/forked-daapd/files/5830237/debug.with.lg.tv.turned.off.log)
[debug turning on lg tv and changing source.log](https://github.com/ejurgensen/forked-daapd/files/5830239/debug.turning.on.lg.tv.and.changing.source.log)

**Comment by ejurgensen** [comment]
Interesting that iOS senders are able to turn the LG on, there must some command or command sequence I am not aware of. It is also interesting that the LG remains in the iOS speaker list, because that it stays in forked-daapd's list is actually a bug that I fixed yesterday (so your build doesn't have it yet). From what I can tell the LG stops announcing, just like if it was powered off. 

If you pull the plug to the LG, does it still stay in the iOS speaker list? I assume it doesn't, so perhaps there is some other "wake up" service that forked-daapd should look for. Could you run the following: `avahi-browse -a -k` and then share the lines with the LG's services?

**Comment by ezerez** [comment]
So I did some more testing. When the TV is turned off the lines
+  ens18 IPv6 LG C8                                         _airplay._tcp        local
+  ens18 IPv4 LG C8                                         _airplay._tcp        local
keep showing up. So I guess its still announcing as a airplay player.
Whats remarkable though is when I unplugged it, it also stayed in my list. even after half an hour it was still there. I tried another airplay box and unplugged it to double check. and that dissapeared after a minute  I even went out to the store checked there. all my airplay devices dissappeaered. When I came back on the wifi, the tv reappeared in my list even though it was still unplugged ...

I must my network might not be the most trustworthy, I also have some homekit devices sometimes not updating. I already checked al my wifi settings that it should be okay, so it could be a homeassistant issue which I use for showing devices in homekit. But it could also be some mdns problems on the wifi. Would be better if someome could doublecheck this behaviour. But anyway it can turn on the TV again, is there anything else I can log to check what it is doing? 

kind regards

**Comment by ejurgensen** [comment]
I don't have a good suggestion on what to check. It could be ordinary wake-on-lan that is used, or it could be something else. I'll leave the issue here, if someone knows how this is done I will be happy to add it to forked-daapd.

**Comment by ezerez** [comment]
Ok too bad. Yes, I think its possible just wake on lan that is actived. The TV also has a tv logo next to it in stead of speaker logo so maybe that makes the difference in the way it connects. If anyone wants me to test something to get this working just leave a message here. Would there be a simple way to send wake on lan command every time I connect to the box? It anyway doesnt do any harm when the TV is on already.

**Comment by ejurgensen** [comment]
You can test wake-on-lan with e.g. etherwake, something like `sudo etherwake -i eth0 AA:BB:CC:DD:EE:FF`, where the argument is the mac address

**Comment by ejurgensen** [comment]
Come to think of it, I'm not sure WOL works via wireless, but you can of course try

**Comment by FrancisHGR** [comment]
WOL works via Wifi, at least on my Sony Bravia TV.

Just fyi: Forked-daapd switches my Yamaha audio receiver on, once I selected the built-in Airplay device and play music via forked-daapd.

**Comment by dmikhalsky** [comment]
LG C8 is Homekit-compatible (at least, my C9 is). So Mac OS might (!) switch it ON via Homekit

**Comment by ejurgensen** [comment]
Yes, sounds likely. Do you happen to know how that works?

**Comment by slappare** [comment]
Hi, did some testing on my Samsung Frame with AirPlay using my macbook that was also showing the output in the airplay list while the TV was off. Captured a start by selecting it as the audio output with Wireshark and it is indeed WOL that is starting the TV.

**Comment by ejurgensen** [comment]
Thanks for investigating. I would think adding WOL isn't too difficult, but an open question is what event OwnTone should then wait for before continuing the start-up sequence (i.e. when is the device ready to accept Airplay requests). Perhaps some mDNS announcement? I will see if I can make a small test program to help figure that out.

**Comment by slappare** [comment]
For the Samsung, it does not show in avahi-browse while turned off. It's announced at wake up (noticeably faster via WOL than via remote) so I guess that would be a good event at least for this TV. I also guess that is why I can't see the output at all in OwnTone while it's turned off...

**Comment by ejurgensen** [comment]
That means that your Samsung works differently than the devices reported above, which isn't so good.

Since avahi-browse doesn't list it when it is turned off, I would also expect that iOS/mac doesn't list it. Isn't that the case?

**Comment by slappare** [comment]
No, it still shows it. Both on iOS and macOS when connected to the wifi.

I noticed that there is an audio output based on the mac adress of the TV which other airplay devices doesn’t have. However I can’t find a way to see how it knows when to list it or not..

**Comment by ejurgensen** [comment]
> I noticed that there is an audio output based on the mac adress of the TV which other airplay devices doesn’t have

Sorry, I don't understand this?

**Comment by ejurgensen** [comment]
What happens if you unplug the Samsung? Is it then listed forever? I think that would be very peculiar.

**Comment by slappare** [comment]
Still there after about 8h unplugged and a reboot of the mac. 

@ezerez, github has formatted you output from avahi-browse, did the lines start with '+' or '-'? Because I believe '-' means that it's no longer "there". And if so, it's the same behaviour as my Samsung. Or did it keep going up and down?
